tests/Unit/ErrorHandling/Test_SegfaultHandler.cpp

```diff
@@ -0,0 +1,22 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <cstdio>
+
+#include "Utilities/ErrorHandling/SegfaultHandler.hpp"
+
+// [[OutputRegex, Segmentation fault!]]
+SPECTRE_TEST_CASE("Unit.ErrorHandling.SegfaultHandler",
+                  "[ErrorHandling][Unit]") {
+  ERROR_TEST();
+  enable_segfault_handler();
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wold-style-cast"
+  // NOLINTNEXTLINE
+  int* foo = (int*)-1;  // make a bad pointer
+#pragma GCC diagnostic pop
+  printf("%d\n", *foo);  // causes segfault
```

Comment:
We may need to do something more clever to trick the optimizer, but we don't need to deal with that unless it becomes an issue.